storage/innobase/srv/srv0start.cc

```diff
@@ -3179,6 +3182,7 @@ innodb_shutdown()
 
 		os_mutex_exit(os_sync_mutex);
 
+	        my_sd_extend_timeout(100000 * 2);
```

Comment:
I would not extend the timeout here. This loop should take virtually no time, because no I/O cannot be done by any of the remaining threads. If this loop executes more than a few rounds, the server must be hung, and we would want systemd to kill us. Given that the data and log files were already written, there should be no harm from the kill.